extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/vpc_flow_log_unmarshaler.go

```diff
@@ -0,0 +1,407 @@
+		return plog.Logs{}, errors.New("still needs to be implemented")
+	default:
+		// not possible, prevent by NewVPCFlowLogUnmarshaler
+		return plog.Logs{}, nil
+	}
+}
+
+type resourceKey struct {
+	accountID string
+	region    string
+}
+
+func (v *vpcFlowLogUnmarshaler) unmarshalPlainTextLogs(reader io.Reader) (plog.Logs, error) {
+	scanner := bufio.NewScanner(reader)
+
+	var fields []string
+	if scanner.Scan() {
+		firstLine := scanner.Text()
+		fields = strings.Split(firstLine, " ")
+	}
+
+	scopeLogsByResource := map[resourceKey]plog.ScopeLogs{}
+	for scanner.Scan() {
+		line := scanner.Text()
+		if err := v.addToLogs(scopeLogsByResource, fields, line); err != nil {
+			return plog.Logs{}, err
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return plog.Logs{}, fmt.Errorf("error reading log line: %w", err)
+	}
+
+	return v.createLogs(scopeLogsByResource), nil
+}
+
+// createLogs based on the scopeLogsByResource map
+func (v *vpcFlowLogUnmarshaler) createLogs(scopeLogsByResource map[resourceKey]plog.ScopeLogs) plog.Logs {
+	logs := plog.NewLogs()
+
+	for key, scopeLogs := range scopeLogsByResource {
+		rl := logs.ResourceLogs().AppendEmpty()
+		attr := rl.Resource().Attributes()
+		attr.PutStr(conventions.AttributeCloudProvider, conventions.AttributeCloudProviderAWS)
+		if key.accountID != "" {
+			attr.PutStr(conventions.AttributeCloudAccountID, key.accountID)
+		}
+		if key.region != "" {
+			attr.PutStr(conventions.AttributeCloudRegion, key.region)
+		}
+		scopeLogs.MoveTo(rl.ScopeLogs().AppendEmpty())
+	}
+
+	return logs
+}
+
+// addToLogs parses the log line and creates
+// a new record log. The record log is added
+// to the scope logs of the resource identified
+// by the resourceKey created from the values.
+func (v *vpcFlowLogUnmarshaler) addToLogs(
+	scopeLogsByResource map[resourceKey]plog.ScopeLogs,
+	fields []string,
+	logLine string,
+) error {
+	// first line includes the fields
+	// TODO Replace with an iterator starting from go 1.24:
+	// https://pkg.go.dev/strings#FieldsSeq
+	nFields := len(fields)
+	nValues := strings.Count(logLine, " ") + 1
+	if nFields != nValues {
+		return fmt.Errorf("expect %d fields per log line, got log line with %d fields", nFields, nValues)
+	}
+
+	// create new key for resource and new
+	// log record to add to the scope of logs
+	// of the resource
+	key := &resourceKey{}
+	record := plog.NewLogRecord()
+
+	// There are 4 fields for the addresses: srcaddr, pkt-srcaddr,
+	// dstaddr, pkt-dstaddr. We will save these fields in a
+	// map, so we can use the right conventions in the end.
+	//
+	// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-records-examples.html#flow-log-example-nat.
+	ips := make(map[string]string, 4)
+
+	start := 0
+	for _, field := range fields {
+		var value string
+		end := strings.Index(logLine[start:], " ")
+		if end == -1 {
+			value = logLine[start:]
+		} else {
+			value = logLine[start : start+end]
+			start += end + 1 // skip the space
+		}
```

Comment:
https://pkg.go.dev/strings#Cut should simplify this, something like:

```go
for _, field := range fields {
	if logLine == "" {
		// handle field count mismatch
	}
	var value string
	value, logLine, _ = strings.Cut(logLine, " ")
}
```

Let's do that in a followup though.

Reply:
Issue so we don't forget: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/39043.